documentation/Using-the-SDK/Advanced/Adding-Column-Variants.md

```diff
@@ -0,0 +1,301 @@
+
+1. Column variants cannot supply a new [column configuration](../../Glossary.md#columnconfiguration) for the column that the variant is for. As such, column variants should only be used when the projected data makes sense given the base column's configuration. If a user would be confused how a column variant's data relates to its base column, you should consider exposing the variant as a separate dedicated column. 
+
+    Note that in some cases it may make sense to edit a column configuration's name and description to accommodate new variants; for example, a column named "Process Name" could be renamed to "Process" and be expose variants for process name, PID, and process name + PID.
+
+2. Column variants cannot be used as a [column role](../../Glossary.md#columnrole) or highlight entries within a [column configuration](../../Glossary.md#columnconfiguration). When column roles or highlight entries are defined, they are always associated with the column's base projection.
+
+# How to Add Column Variants
+
+Column variants are added to a column by adding the base column to an `ITableBuilder` with the `AddColumnWithVariants` methods introduced in SDK version `1.3`. These methods take a callback `Func<RootColumnBuilder, ColumnBuilder>` whose return value defines all the variants to associate with the added column. The supplied `Func` will be invoked with a `RootColumnBuilder` supplied by the SDK runtime and must return the final `ColumnBuilder` configured with variants.
+
+Note that unlike the `ITableBuilder`, column builders are purely functional components: every column builder method returns a **new instance** of a column builder. For example, consider the code snippet below:
+
+```cs
+tableBuilder
+    .AddColumnWithVariants(baseConfig, projectionA, builder =>
+    {
+        // These modes are ignored because the return value from
+        // WithMode(modeB, projectionB) is being discarded 
+        builder
+            .WithModes("Mode A")
+            .WithMode(modeB, projectionB);
+
+        // Will not contain the modes added above because the returned value is building
+        // off of the input builder
+        return builder
+                .WithToggle(toggleIdentifier, projectionToggle);
+    });
+```
+
+In this example, the added column will only contain the base column with one toggle. **There will not be any modes associated with this column.** Even though `builder.WithModes` was called, the `ColumnBuilder` returned by the method was the supplied (empty) builder with one `WithToggle` added.
+
+Any time a new projection is associated with a variant, a `ColumnVariantDescriptor` must be used to uniquely identify the added variant amongst all variants in the column. The `ColumnVariantDescriptor` is created either
```

Comment:
Thoughts on making this a small dedicated `# Column Variant Descriptors` section?

Reply:
There is already a lot of content here. Unless it's very important, I'd prefer deferring it to the class summary